### ISSUE TITLE
Site_prism started to throw uninitialized constant SitePrism::Section error

### ISSUE THREAD

**Issue by ekinciq** [site-prism/site_prism#6]
## Environment
- Operating System ( Windows)
- What Gems / Versions you are using 
site_prism v 3.3 
 capybara (~> 3.25.0)
- I have Cucumber tests
- Browser/s Driver/s being used - headless chrome on Docker container or local terminal 

## Expected Behavior
site_prism v 3.3 installed and working as before

## Actual Behavior
site_prism v 3.3 is auto installed with bundle_install but now gives following error when running tests: 
uninitialized constant SitePrism::Section
Did you mean?  Exception (NameError)
![image](https://user-images.githubusercontent.com/10047730/60519735-0b6de380-9cdc-11e9-92d5-a2e2406fca5e.png)

it ran successfully last time at 01 Jul 2019 at 13:39:38 BST

## Steps to reproduce
I didn't change anything on my code base, it used to work fine with site_prism (3.2) but after bundle install now I have  site_prism (3.3) and site_prism-all_there (~> 0.1.0)  site_prism-all_there (0.1) in my Gemfile.lock and my tests doesn't run anymore and throws below error. 

```
Headers_and_footers_rb in shared sections:
class Header < SitePrism::Section
  set_default_search_arguments '.header'
  element :provider_name, '.header__provider-name'
  element :provider_logo, '.header__service-logo'
end

admin_homepage.rb that uses headers_and_footers 
require 'capybara'
require 'capybara/cucumber'
require 'selenium-webdriver'
require 'site_prism'
require_relative '../page_objects/shared_sections/headers_and_footers'
require_relative '../page_objects/shared_sections/navigation_bars'

class AdminHomePage < SitePrism::Page
  set_url '/home.htm'
  section :header, Header
  section :footer, Footer
end

and then calling it from .rb
When(/^Admin clicks allocate cases button$/) do
  @home = AdminHomePage.new
  @home.quick_links.allocate_cases_link.click
end
```

**Comment by luke-hill** [comment]
Can you follow the template please, as it's designed to help. I cannot triage this.

**Comment by ekinciq** [comment]
sorry @luke-hill what template?

**Comment by luke-hill** [comment]
When you raise a new issue, there are instructions in there. Can you make sure you fill it out properly. You've deleted the portion that asks you to provide an SSCCE, ideally in the form of a github repo or suchlike.

Without a reproducible case I cannot triage it. The test suite runs (cucumber/capybara/site_prism), on github so it's not globally broken.

Having released a new co-dependent gem something small may have slipped through, but I need to see some code examples first, not just, it doesn't work.

Thankyou

**Comment by ekinciq** [comment]
updated the ticket. Sorry

**Comment by luke-hill** [comment]
Oki I've had a quick go, but as I mentioned, this isn't a reproducible test case....

It works for me for what I've put together.

Can you please re-read what we've asked for and the links provided. Ideally provide the code in the form of a github repo. One which can be cloned and ran.

**Comment by luke-hill** [comment]
Were you able to get it working locally? Just would be good to note for others.

**Comment by ekinciq** [comment]
sorry no but I am going to try to use the working version. I can't really give more code from my code base

**Comment by luke-hill** [comment]
I'm not asking you to give proprietary code.

From the code you've given can you just create a standalone repo on github? It shouldn't take you more than 5mins. Or better yet code it against something like https://the-internet.herokuapp.com/

**Comment by ekinciq** [comment]
it is really an interesting issue. It works fine when I run tests on Rubymine but it doesn't if I run via console application. I will try to solve and let you know if I can get anywhere. It is not really related a code block my whole project stopped working on Gocd

**Comment by ekinciq** [comment]
I think I might found the issue. when you call `require 'site_prism'` there are now two site_prism namespace 
![image](https://user-images.githubusercontent.com/10047730/60595741-8ba94d00-9d9f-11e9-95ad-2bd64fb8a2f1.png)

Can this cause a kind of race condition or unknown of the namespace? 
when I require with namespace like require '`C:\tools\ruby26\lib\ruby\gems\2.6.0\gems\site_prism-3.3\lib\site_prism.rb' `it started to work fine for me! 

I can't commit this change since we don't have same ruby version for my local and docker agent. Is there any better solution that you can suggest? @luke-hill

**Comment by luke-hill** [comment]
Right, this was something I found out yesterday myself when I started writing the rest of my tests with co-dependencies.

The fact there are 2 copies of the namespace won't affect this, but the filenames possibly could yes. I've gone about moving the `all_there` config file from an identically named file to one that is different.

Have you managed to produce a reproducible test case though showing the issue?
